File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    void backtrack(vector<vector<int>>&ans, vector<int>& nums, int start, vector<int>&list){
        // if(start>list.size()){
            ans.push_back(list);
            // return;
        // }
        for(int i=start;i<nums.size();i++){
            list.push_back(nums[i]);
            backtrack(ans,nums,i+1,list);
            list.pop_back();
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>ans;
        vector<int>list;
        backtrack(ans,nums,0,list);
        return ans;
    }
};

File: 0078-subsets/0078-subsets.java
class Solution {
    public void helper(int[] nums, List<List<Integer>> ans, List<Integer>list, int idx){
        if(idx == nums.length){
            ans.add(new ArrayList<>(list));
            return;
        }
        list.add(nums[idx]);
        helper(nums, ans, list, idx+1);
        list.remove(list.size()-1);
        helper(nums, ans, list, idx+1);
    }
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> ans = new ArrayList<>();
        helper(nums, ans, new ArrayList<>(), 0);
        return ans;
    }
}